### ISSUE TITLE
how to animate list on populate

### ISSUE THREAD

**Issue by abhi98228** [wasabeef/recyclerview-animators#9]
How do I animate the list when my activity starts and list populates

**Comment by faizsiddiqui** [comment]
Try this.. only AlphaInAnimationAdapter working in this case. I too facing this issue.
    AlphaInAnimationAdapter alphaAdapter = new AlphaInAnimationAdapter(mAdapter);
    alphaAdapter.setDuration(1000);
    mRecyclerView.setAdapter(alphaAdapter);

**Comment by faizsiddiqui** [comment]
Try this.. only AlphaInAnimationAdapter working in this case. I too facing this issue.

    AlphaInAnimationAdapter alphaAdapter = new AlphaInAnimationAdapter(mAdapter);
    alphaAdapter.setDuration(1000);
    mRecyclerView.setAdapter(alphaAdapter);

**Comment by wasabeef** [comment]
What you have set?

**Comment by faizsiddiqui** [comment]
CardView with images, some text.
![screenshot_1](https://cloud.githubusercontent.com/assets/3103741/6545498/40a89368-c5b0-11e4-93a4-0ce81e66e9fd.png)

**Comment by wasabeef** [comment]
Yes. please show me the code.

**Comment by faizsiddiqui** [comment]
public class CardView extends RecyclerView.Adapter<CardView.ViewHolder> {

		public String[] titles, descriptions, images;
		OnItemClickListener mItemClickListener;
		ImageLoader mImageLoader;

		public CardView(String[] titles, String[] descriptions, String[] images) {
			this.titles = titles;
			this.descriptions = descriptions;
			this.images = images;
		}

		@Override
		public CardView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
			View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.card_view, parent, false);
			mImageLoader = VolleySingleton.getInstance().getImageLoader();
			return new ViewHolder(view);
		}

		@Override
		public void onBindViewHolder(CardView.ViewHolder holder, int position) {
			holder.cardViewTitle.setText(titles[position]);
			holder.cardViewText.setText(descriptions[position]);
			holder.cardNetworkImageView.setImageUrl(images[position], mImageLoader);
		}

		@Override
		public int getItemCount() {
			return titles.length;
		}

		public class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {

			NetworkImageView cardNetworkImageView;
			TextView cardViewTitle, cardViewText;

			public ViewHolder(View itemView) {
				super(itemView);
				cardNetworkImageView = (NetworkImageView) itemView.findViewById(R.id.cardViewImage);
				cardViewTitle = (TextView) itemView.findViewById(R.id.cardViewTitle);
				cardViewText = (TextView) itemView.findViewById(R.id.cardViewText);
				itemView.setOnClickListener(this);
			}

			@Override
			public void onClick(View v) {
				if (mItemClickListener != null) {
					mItemClickListener.onItemClick(v, getPosition());
				}
			}
		}

		public interface OnItemClickListener {
			public void onItemClick(View view, int position);
		}

		public void SetOnItemClickListener(OnItemClickListener mItemClickListener) {
			this.mItemClickListener = mItemClickListener;
		}

	}

**Comment by hbb20** [comment]
@abhi98228 and @faizsiddiqui  ,Let me understand your question. I think you can see the animation while scrolling, but you also want to see the animation when the activity is started? Right?(No?, please make me clear) but If so, let me try to help. If your list data is already loaded. (I means when you do not need to make http call to receive list data). In that case ,Animation would be fast enough so that you might not notice them. Please try inside your onViewCreate() or equivalent method (where you assign view object from XML file's rootView.)

     mRecyclerView=(RecyclerView)rootView.findViewById(R.id.myRecyclerView);
     recyclerView.post(new Runnable() {
                @Override
                public void run() {
	                mAdapter = new CardView(titles);
					AlphaInAnimationAdapter alphaAdapter = new AlphaInAnimationAdapter(mAdapter);
					mRecyclerView.setAdapter(alphaAdapter);      
                }
            });
If still its fast, try by adding timer delay.
Check [What does post(runnable) do?](http://goo.gl/r7IkOp)

**Comment by faizsiddiqui** [comment]
@hbb20 No, I'm not even seeing animation on Scrolling. Only "AlphaInAnimationAdapter" working.

**Comment by dongseok0** [comment]
Yeah same here. only AlphaInAnimationAdapter work.
